test/boost/secondary_index_test.cc

```diff
@@ -1529,26 +1550,315 @@ SEASTAR_TEST_CASE(test_token_order) {
             cquery_nofail(e, format("INSERT INTO t (pk, v) VALUES ({}, 1)", i).c_str());
         }
 
-        std::vector<std::vector<bytes_opt>> expected_rows = {
-            { int32_type->decompose(5) }, // token(pk) = -7509452495886106294
-            { int32_type->decompose(1) }, // token(pk) = -4069959284402364209
-            { int32_type->decompose(0) }, // token(pk) = -3485513579396041028
-            { int32_type->decompose(2) }, // token(pk) = -3248873570005575792
-            { int32_type->decompose(4) }, // token(pk) = -2729420104000364805
-            { int32_type->decompose(6) }, // token(pk) = +2705480034054113608
-            { int32_type->decompose(3) }, // token(pk) = +9010454139840013625
-        };
-
         eventually([&] {
             auto nonindex_order = cquery_nofail(e, "SELECT pk FROM t");
             auto index_order = cquery_nofail(e, "SELECT pk FROM t WHERE v = 1");
 
-            assert_that(nonindex_order).is_rows().with_rows(expected_rows);
-            assert_that(index_order).is_rows().with_rows(expected_rows);
+            assert_that(nonindex_order).is_rows().with_rows(testset_pks);
+            assert_that(index_order).is_rows().with_rows(testset_pks);
         });
     });
 }
 
+SEASTAR_TEST_CASE(test_select_with_token_range_cases) {
+    return do_with_cql_env_thread([] (auto& e) {
+        cquery_nofail(e, "CREATE TABLE t (pk int, v int, PRIMARY KEY(pk))");
+
+        for (int i = 0; i < 7; i++) {
+            cquery_nofail(e, format("INSERT INTO t (pk, v) VALUES ({}, 1)", i).c_str());
+        }
+
+        auto get_result_rows = [&](int start, int end) {
+            std::vector<std::vector<bytes_opt>> result;
+            for (int i = start; i <= end; i++) {
+                result.push_back({ testset_pks[i][0], int32_type->decompose(1) });
+            }
+            return result;
+        };
+
+        auto do_tests = [&] {
+            // inclusive-inclusive range
+            auto msg = cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND token(pk) >= {} AND token(pk) <= {} ALLOW FILTERING", testset_tokens[1], testset_tokens[4]).c_str());
```

Comment:
This pattern repeats so much that I think it should be DRYd.  Something like:
```c++
auto q = [&] (sstring_view token_restriction) { return cquery_nofail(e, format("SELECT * FROM t WHERE v = 1 AND {} ALLOW FILTERING", token_restriction); }

auto closed_range = [] (int a, int b) { return format("token(pk) <= {} AND token(pk) <= {}", a, b); }
auto open_range = [] (int a, int b) { return format("token(pk) < {} AND token(pk) < {}"; }
auto left_half_open_range = [] (int a, int b) { return format("token(pk) < {} AND token(pk) <= {}"; }
auto left_half_open_range = [] (int a, int b) { return format("token(pk) <= {} AND token(pk) < {}"; }

assert_that(q(closed_range(testset_tokens[1], testset_tokens[4])))...
```
This makes it more readable (to me, at least), and you don't need the comments because the function names tell us what's being tested.